packages/@aws-cdk/example-construct-library/README.md

```diff
@@ -0,0 +1,88 @@
+## An example Construct Library module
+
+<!--BEGIN STABILITY BANNER-->
+---
+
+![cfn-resources: Stable](https://img.shields.io/badge/cfn--resources-stable-success.svg?style=for-the-badge)
+
+![core-constructs: Stable](https://img.shields.io/badge/cdk--constructs-stable-success.svg?style=for-the-badge)
+
+---
+<!--END STABILITY BANNER-->
+
+This package contains an example CDK construct library
+for an imaginary resource called ExampleResource.
+Its target audience are construct library authors -
+both when contributing to the core CDK codebase,
+or when writing your own construct library.
+
+Even though different construct libraries model vastly different services,
+a large percentage of the structure of the construct libraries
+(what we often call Layer 2 constructs, or L2s for short)
+is actually strikingly similar between all of them.
+This module hopes to present a skeleton of that structure,
+that you can literally copy&paste to your own construct library,
+and then edit to suit your needs.
+It also attempts to explain the elements of that skeleton as best as it can,
+through inline comments on the code itself.
+
+## Using when contributing to the CDK codebase
+
+If you're creating a completely new module,
+feel free to copy&paste this entire directory,
+and then edit the `package.json` and `README.md`
+files as necessary (see the "Package structure" section below).
+Make sure to remove the `"private": true` line from `package.json`
+after copying, as otherwise your package will not be published!
+
+If you're contributing a new resource to an existing package,
+feel free to copy&paste the following files,
+instead of the entire package:
+
+* [`lib/example-resource.ts`](lib/example-resource.ts)
+* [`lib/example-resource-utils.ts`](lib/example-resource-utils.ts)
+* [`test/example-resource.test.ts`](test/example-resource.test.ts)
+* [`test/integ.example-resource.ts`](test/integ.example-resource.ts)
+* [`test/integ.example-resource.expected.json`](test/integ.example-resource.expected.json)
+
+And proceed to edit and rename them from there.
+
+## Using for your own construct libraries
+
+Feel free to use this package as the basis of your own construct library;
+note, however, that you will have to change a few things in `package.json` to get it to build:
+
+* You probably want to remove `"private": true`, otherwise you won't be able to publish your package.
```

Comment:
"probably want to"? Be more clear - 'remove the flag if you intend to publish the package to npm' and provide link to https://docs.npmjs.com/files/package.json#private

Reply:
And you're sure we correctly use this flag in all our tooling? Would it be better to put this in a directory like `tools/` that we never expect to publish? 

The risk I'm seeing is a standard bulk edit, such as what we did for eslint, jest, etc. that accidentally switches this on and we don't realize it. Not a big risk but would definitely be very confusing to CDK customers.

Reply:
> "probably want to"? Be more clear - 'remove the flag if you intend to publish the package to npm' and provide link to https://docs.npmjs.com/files/package.json#private

Done.

Reply:
> And you're sure we correctly use this flag in all our tooling? Would it be better to put this in a directory like `tools/` that we never expect to publish?
> 
> The risk I'm seeing is a standard bulk edit, such as what we did for eslint, jest, etc. that accidentally switches this on and we don't realize it. Not a big risk but would definitely be very confusing to CDK customers.

I'm fairly confident in this flag. I believe we don't have any linters that error out on this flag (as the package builds), and also we don't generally have global executions of fix violations for all packages at the same time.

I'd rather not put this in `tools/`, as that's a poor experience for an example package I feel.